drivers/gpio/gpio_gecko.c

```diff
@@ -154,13 +154,21 @@ static int gpio_gecko_write(struct device *dev,
 			 * Writing zeros to the other bits leaves the data
 			 * output unchanged for the other pins.
 			 */
+#if defined(_GPIO_P_DOUTSET_MASK)
 			gpio_base->DOUTSET = BIT(pin);
+#else
+			BUS_RegBitWrite(&gpio_base->DOUT, pin, 1);
+#endif
 		} else {
 			/* Clear the data output for the corresponding pin.
 			 * Writing zeros to the other bits leaves the data
 			 * output unchanged for the other pins.
 			 */
+#if defined(_GPIO_P_DOUTCLR_MASK)
 			gpio_base->DOUTCLR = BIT(pin);
+#else
+			BUS_RegBitWrite(&gpio_base->DOUT, pin, 0);
```

Comment:
I think you can use a more clear assignment like:
gpio_base->DOUT &= ~BIT(pin);